app/uk/gov/hmrc/nisp/services/NationalInsuranceService.scala

```diff
@@ -53,7 +55,7 @@ class NationalInsuranceService @Inject()(nationalInsuranceConnector: NationalIns
 
         case Left(errorResponse) => Left(errorResponse)
 
-        case value => throw new NotImplementedError(s"Match not implemented for: $value")
+        case value => Left(UpstreamErrorResponse(s"Match not implemented for: $value", serverInternalError))
```

Comment:
is there a unit test for this because I feel like you may have changed the behaviour slightly here.

Reply:
No. The execution was never tested if it was that test would now fail.  I suspect because all responses from the service are included in the match but the compiler thinks some other value is possible. How would you test this?

Reply:
so previously this would throw an exception, but now it returns an error response. Is that being handled by whatever calls this?

Reply:
I can not find any catch for that exception in the code. Assumption is that it would be caught by a global exception handler that would produce an appropriate response to the user and log the issue.
No test unit, integration or acceptance  (I ran all) failed so no test exists for this.
A unit test would only prove that it returned the expected result but the more important would be an integration test which would require the getNationalInsurance method to return a value not matching the other cases and dont know what that would be. 
I suspect in reality the exception would was never thrown and was added to complete the match.

Who would know other wise?